server/src/inmantals/server.py

```diff
@@ -238,21 +256,33 @@ def _get_module_name():
 
             raise InvalidExtensionSetup(error_message)
 
-        metadata: typing.Mapping[str, object] = {
-            "name": "Temporary project",
-            "description": "Temporary project",
-            "modulepath": "libs",
-            "downloadpath": "libs",
-            "install_mode": install_mode.value,
-        }
-        if self.handler.repos:
-            metadata["repo"] = self.handler.repos
+        def _generate_project_yml() -> dict[str, object]:
+            """
+            Generate the content of the project.yml for the temporary project as a python dict.
+            """
+            content = {
+                "name": "Temporary project",
+                "description": "Temporary project",
+                "modulepath": "libs",
+                "downloadpath": "libs",
+            }
+            if SUPPORTS_PROJECT_PIP_INDEX:
+                content["pip"] = self.handler.pipconfig
+            else:
+                content["install_mode"]: install_mode.value
+                if self.handler.repos:
+                    content["repo"] = self.handler.repos
+
+            return content
+
         logger.debug(
-            "project.yaml created at %s, repos=%s", os.path.join(inmanta_project_dir, "project.yml"), self.handler.repos
+            "project.yml created at %s, repos=%s, pip_config=%s",
+            os.path.join(inmanta_project_dir, "project.yml"),
+            self.handler.repos,
+            self.handler.pipconfig,
```

Comment:
Let's add the following assertion to the constructor of this class: `self.handler.pipconfig is not None`